examples/esp-todomvc-react/js/model/eventProcessors/TodoListPostEventProcessor.js

```diff
@@ -0,0 +1,49 @@
+/*global app*/
+
+(function () {
+    "use strict";
+
+    app.model.TodoListPostEventProcessor = {
+        process: function (model) {
+            var filter,
+                id,
+                filteredTodoItems = [],
+                incompleteCount = 0,
+                completeCount = 0,
+                count = 0,
+                allChecked = true,
+                todoItem;
+            if (model.footer.filter === app.model.Filter.complete) {
+                filter = function (todoItem) { return todoItem.complete; };
+            }
+            else if (model.footer.filter === app.model.Filter.active) {
+                filter = function (todoItem) { return !todoItem.complete; };
+            } else {
+                filter = function () { return true; };
+            }
+            for (id in model.todoItemsById) {
+                if (model.todoItemsById.hasOwnProperty(id)) {
+                    todoItem = model.todoItemsById[id];
+                    if (filter(todoItem)) {
+                        filteredTodoItems.push(todoItem);
+                        allChecked = todoItem.complete && allChecked;
+                    }
+
+                    if (!todoItem.complete) {
+                        incompleteCount++;
+                    } else {
+                        completeCount++;
+                    }
+
+                    count++;
+                }
+            }
+            model.main.filteredTodoItems = filteredTodoItems;
+            model.main.toggleAll.visible = filteredTodoItems.length > 0;
```

Comment:
> Generally you don't want the UI state changing outside the router.

It wouldn't be, it is UI derived directly off model state. Personally I see a lot more value in esp modelling the domain and metadata which the UI can use to render however it wants.